repose-aggregator/services/rate-limiting-service/src/main/resources/META-INF/schema/config/rate-limiting-configuration.xsd

```diff
@@ -162,6 +162,10 @@
                    xerces:message="Only one default limit group may be defined"
                    saxon:message="Only one default limit group may be defined"/>
 
+        <xs:assert vc:minVersion="1.1" test="count(distinct-values(limit-group/limit/@id)) = count(limit-group/limit/@id)"
+                   xerces:message="ALL Limits must have unique ids across all limit groups"
+                   saxon:message="ALL Limits must have unique ids across all limit groups" xpathDefaultNamespace="##targetNamespace"/>
+
         <!-- TODO: assert at least one limit-group or global-limit-group is present -->
```

Comment:
The to-do was there before I added the xsd assertion by the way. I didn't write that. 

On October 21, 2015 1:47:44 PM CDT, William Scheidegger <notifications@github.com> wrote:
>> @@ -162,6 +162,10 @@
>>                     xerces:message="Only one default limit group may
>be defined"
>>                     saxon:message="Only one default limit group may
>be defined"/>
>>  
>> +        <xs:assert vc:minVersion="1.1"
>test="count(distinct-values(limit-group/limit/@id)) =
>count(limit-group/limit/@id)"
>> +                   xerces:message="ALL Limits must have unique ids
>across all limit groups"
>> +                   saxon:message="ALL Limits must have unique ids
>across all limit groups" xpathDefaultNamespace="##targetNamespace"/>
>> +
>>          <!-- TODO: assert at least one limit-group or
>global-limit-group is present -->
>
>Easy enough.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/rackerlabs/repose/pull/1398/files#r42665423

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.